### PR TITLE
[GPU] Do not add paddings for OneDNN primitives

### PR DESCRIPTION
**Pull request by sshlyapn**
### Details:
 - Prohibit paddings for OneDNN primitives
 - Fix review comments for PR (#7727)

### CODE REVIEW

**Review by byungilm — APPROVED** [review]

**Review by lznamens — APPROVED** [review]
LGTM.

File: inference-engine/thirdparty/clDNN/tests/test_cases/convolution_gpu_test.cpp [hunk 1]
```diff
@@ -9066,4 +9066,81 @@ TEST_P(convolution_gpu_onednn, conv_onednn_cases) {
                 }
 }
 
+TEST(convolution_gpu_onednn, padding_for_cldnn_kernel_after_onednn) {
```
Comment (isanghao) [inline inference-engine/thirdparty/clDNN/tests/test_cases/convolution_gpu_test.cpp:9069]: nice test case!

File: inference-engine/thirdparty/clDNN/src/graph_optimizer/prepare_padding.cpp [hunk 2]
```diff
@@ -17,10 +17,30 @@ void prepare_padding::run(program& p) {
...
+                auto& input = node.get_dependency(0);
+                if (input.get_preferred_impl_type() == impl_types::onednn &&
+                    node.get_preferred_impl_type() == impl_types::ocl &&
+                    static_cast<bool>(needed_padding)) {
+                    auto new_reorder = std::make_shared<reorder>(node.id() + "_padding_reorder_for_" + input.id(), input.id(), input.get_output_layout());
+                    auto& new_reorder_node = p.get_or_create(new_reorder);
+                    p.add_intermediate(new_reorder_node, node, input);
+                }
+
+                p.apply_needed_padding(node, node.get_dependency(0), needed_padding);
```
Comment (isanghao) [inline inference-engine/thirdparty/clDNN/src/graph_optimizer/prepare_padding.cpp:41]: Could you add an assertion in cldnn conv itself to confirm that it has proper padding for input and output?

**Review by isanghao — DISMISSED** [review]
Could you add an assertion for more clarify of padding behavior? https://github.com/openvinotoolkit/openvino/pull/8619/files#r751803079

File: inference-engine/thirdparty/clDNN/src/graph_optimizer/prepare_padding.cpp (hunk 2 above)
Comment (isanghao) [reply inference-engine/thirdparty/clDNN/src/graph_optimizer/prepare_padding.cpp:41]: it is always there. so mark it as resolved.

**Pull request merged**